File: client/app/schemas/cards/video_card.js
const { required } = require('../../helpers/validations')
const { extend } = require('../../helpers/utilities')
const cardSchema = require('../card')

module.exports = extend({}, cardSchema, {
  'data.video_id': {
    type: 'text',
    validations: [required],
  },
  'data.site': {
    type: 'select',
    validations: [required],
    options: [{ value: 'youtube' }, { value: 'vimeo' }],
  },
})

File: server/schemas/cards/video_card.py
from modules.util import extend
from modules.validations import is_required, is_string, is_one_of
from schemas.card import schema as card_schema


schema = extend({}, card_schema, {
  'fields': {
    'data': {
      'embed': {
        'site': {
          'validate': (is_required, is_string, (is_one_of, 'youtube', 'vimeo'),),
        },
        'video_id': {
          'validate': (is_required, is_string,),
        },
      },
    },
  },
})
